### ISSUE TITLE
fix: (Regression) ExpandedChanged binding in FluentNavGroup no longer firing

### ISSUE THREAD

**Issue by teyc** [microsoft/fluentui-blazor#2289]
# 🐛 Bug Report

Versions affected: 4.8.0, 4.81
Version not affected: 4.7.2

## 💻 Repro or Code Sample

```
   <nav class="sitenav">
    <FluentNavMenu Id="bug-menu">
        <FluentNavGroup @bind-Expanded="expanded">
            <TitleTemplate>
                Sample Title @expanded
            </TitleTemplate>
        </FluentNavGroup>
    </FluentNavMenu>
    </nav>
</div>

@code {
    private bool expanded = true;
}
```

## 🤔 Expected Behavior
Clicking should toggle value
![image](https://github.com/microsoft/fluentui-blazor/assets/2076/09307515-289b-4999-8c06-3198ab8ee102)
![image](https://github.com/microsoft/fluentui-blazor/assets/2076/3395e49d-5b32-438a-a689-44d7870b5001)

## 😯 Current Behavior
Clicking doesn't toggle value
![image](https://github.com/microsoft/fluentui-blazor/assets/2076/09307515-289b-4999-8c06-3198ab8ee102)
![image](https://github.com/microsoft/fluentui-blazor/assets/2076/2b83b768-4592-404b-a864-c24a4d18f2ee)

## 💁 Possible Solution
Untested https://github.com/microsoft/fluentui-blazor/compare/dev...teyc:fix/2289-ExpandedChanged?expand=1
Related to https://github.com/microsoft/fluentui-blazor/commit/d0a0a4edf67dd98b0a805e265e5e10cc589a4b6f

## 🔦 Context

We use the `Expanded` state to virtualise child content, as we have a lot of data.

## 🌍 Your Environment

Versions affected: 4.8.0, 4.81
Version not affected: 4.7.2

**Comment by vnbaaij** [comment]
Is this Server or WebAssembly?

**Comment by vnbaaij** [comment]
This is probably going to be difficult to replicate. Can you supply a zip of a project or a repo to work against?

**Comment by teyc** [comment]
see possible fix. I couldn't build fluentui-blazor 
https://github.com/microsoft/fluentui-blazor/compare/dev...teyc:fix/2289-ExpandedChanged?expand=1
This regression is caused by `Expanded` property being set too early, and causes the change detection to not fire the value has changed.

**Comment by ASchweers** [comment]
Same issue here. But only the first FluentNavGroup-Element is affected by this (in the demo-app it's the "Getting Started"-NavGroup).

**Comment by vnbaaij** [comment]
> [Issue-2289.zip](https://github.com/user-attachments/files/16024898/Issue-2289.zip) see possible fix. I couldn't build fluentui-blazor https://github.com/microsoft/fluentui-blazor/compare/dev...teyc:fix/2289-ExpandedChanged?expand=1 This regression is caused by `Expanded` property being set too early, and causes the change detection to not fire the value has changed.

Your example code is faulty. The NavMenu component is missing and the project is using SSR so Expanded will never be set (needs interactivity)

**Comment by teyc** [comment]
[Issue-2289.zip](https://github.com/user-attachments/files/16028740/Issue-2289.zip)
Thank you for looking at this so quickly!

Apologies I was cleaning up the bug to remove irrelevant parts and I didn't know what I was doing. This one pins to 4.8.0, and you can see that in `Home.razor` the "Expanded = false" is sticky, while if you revert to 4.7.2, you can see the "Expanded = true" toggles to "Expanded = false"

**Comment by teyc** [comment]
Thank you for looking at this so quickly! I had accidentally removed too much code when I cleaned up the sample.

Please have a look at Home.razor and observe the `Expanded =` when you switch from Version 4.8.0 to Version 4.7.2
[Issue-2289.zip](https://github.com/user-attachments/files/16028793/Issue-2289.zip)

**Comment by vnbaaij** [comment]
Ok, we are going back to the 4.7.2 code for the next release. With that this issue will be solved and fixed. 

We made a change to accommodate working with `prerender: false` in the `Route`s rendermode. When prerendering is false, the page script component gets triggered and expanding/collapsing is actually being done by script instead of by the interactive code. The problem you are seeing is a side-effect of that in combination with the changes we made.
